src/test/java/yablunovskyi/bookstore/controller/BookControllerTests.java

```diff
@@ -0,0 +1,398 @@
+    
+    @BeforeAll
+    static void beforeAll(
+            @Autowired WebApplicationContext applicationContext,
+            @Autowired DataSource dataSource) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        //teardown(dataSource);
+        //setUpCategories(dataSource);
+    }
+    
+    /*@AfterAll
+    static void afterAll(@Autowired DataSource dataSource) {
+        teardownCategories(dataSource);
+    }
+    
+    @BeforeEach
+    void beforeEach(@Autowired DataSource dataSource) {
+        setUpBooks(dataSource);
+    }
+    
+    @AfterEach
+    void afterEach(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
+    
+   @SneakyThrows
+    private static void setUpCategories(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/categories/add-default-categories.sql")
+            );
+        }
+    }
+    
+    @SneakyThrows
+    private static void setUpBooks(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/add-default-books.sql")
+            );
+        }
+    }
+    
+    @SneakyThrows
+    private static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/delete-all-books.sql")
+            );
+        }
+    }
+    
+    @SneakyThrows
+    private static void teardownCategories(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/categories/delete-all-categories.sql")
+            );
+        }
+    }*/
+    
+    @Test
+    @DisplayName("""
+            Verify save() method works and save a new book to database""")
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    public void save_ValidRequestDto_Success() throws Exception {
+        //Given
+        BookRequestDto requestDto = new BookRequestDto(
+                "title",
+                "author",
+                Set.of(1L, 2L),
+                "1234567800",
+                BigDecimal.valueOf(400),
+                "description",
+                "link"
+        );
+        BookResponseDto expected = new BookResponseDto();
```

Comment:
can you use a constructor? Also better to move it to a private method and simplify  your test